felix/fv/donottrack_test.go

```diff
@@ -38,6 +39,196 @@ import (
+				Expect(err).NotTo(HaveOccurred())
+				lines := strings.Split(strings.Trim(defaultRoute, "\n "), "\n")
+				Expect(lines).To(HaveLen(1))
+				defaultRouteArgs := strings.Split(strings.Replace(lines[0], "eth0", "bond0", -1), " ")
+
+				// Assuming the subnet route will be "proto kernel" and that will be the only such route.
+				subnetRoute, err := felix.ExecOutput("ip", "route", "show", "proto", "kernel")
+				Expect(err).NotTo(HaveOccurred())
+				lines = strings.Split(strings.Trim(subnetRoute, "\n "), "\n")
+				Expect(lines).To(HaveLen(1), "expected only one proto kernel route, has docker's routing set-up changed?")
+				subnetArgs := strings.Split(strings.Replace(lines[0], "eth0", "bond0", -1), " ")
+
+				felix.Exec("ip", "addr", "del", felix.IP, "dev", "eth0")
+				felix.Exec("ip", "link", "add", "dev", "bond0", "type", "bond")
+				felix.Exec("ip", "link", "set", "dev", "eth0", "down")
+				felix.Exec("ip", "link", "set", "dev", "eth0", "master", "bond0")
+				felix.Exec("ip", "link", "set", "dev", "eth0", "up")
+				felix.Exec("ip", "link", "set", "dev", "bond0", "up")
+
+				ipWithSubnet := felix.IP + "/" + felix.GetIPPrefix()
+				felix.Exec("ip", "addr", "add", ipWithSubnet, "dev", "bond0")
+				felix.Exec(append([]string{"ip", "r", "add"}, defaultRouteArgs...)...)
+				felix.Exec(append([]string{"ip", "r", "replace"}, subnetArgs...)...)
+				if BPFMode() {
+					ensureRightIFStateFlags(felix, ifstate.FlgIPv4Ready, ifstate.FlgBondSlave, map[string]uint32{"bond0": ifstate.FlgIPv4Ready | ifstate.FlgBond})
+				}
+				createHostEndpoint(felix, "bond0", []string{felix.IP}, client, ctx)
+			}
+		})
+
+		It("should implement untracked policy correctly", func() {
+			// This test covers both normal connectivity and failsafe connectivity.  We combine the
+			// tests because we rely on the changes of normal connectivity at each step to make sure
+			// that the policy has actually flowed through to the dataplane.
+
+			By("having only failsafe connectivity to start with")
+			expectFailSafeOnlyConnectivity(tc, hostW, externalClient, cc)
+			By("Having no Linux IP sets")
+			Consistently(tc.Felixes[0].IPSetNames, "2s", "1s").Should(BeEmpty())
+			host0Selector := fmt.Sprintf("name == 'bond0-%s'", tc.Felixes[0].Name)
+			host1Selector := fmt.Sprintf("name == 'bond0-%s'", tc.Felixes[1].Name)
+
+			By("Having connectivity after installing bidirectional policies")
+			host0Pol := createDonotTrackPolicy("host-0-pol", host0Selector, host1Selector, client, ctx)
+			_ = createDonotTrackPolicy("host-1-pol", host1Selector, host0Selector, client, ctx)
+
+			Consistently(xdpProgramAttached(tc.Felixes[0], "bond0"), "2s", "1s").Should(BeFalse())
+			Consistently(xdpProgramAttached(tc.Felixes[1], "bond0"), "2s", "1s").Should(BeFalse())
+			Eventually(xdpProgramAttached(tc.Felixes[0], "eth0"), "10s", "1s").Should(BeTrue())
+			Eventually(xdpProgramAttached(tc.Felixes[1], "eth0"), "10s", "1s").Should(BeTrue())
```

Comment:
In a way this is the only thing that differs from the non-bond tests. Can't you just place the test in a function and parametrize it whether it is with or w/o bond? Connectivity should always be the same, but where the programs are attached is different.